### PR TITLE
Feature/argo2ioda v2

### LINKED ISSUE #705: Upgrade Python converter for argoClim2ioda to IODA-v2:
Update the converter identified in the title (hereafter referred to as the **target converter**) to use the IODA-v2 API.

- Update import statement
- Variable dimensions dictionary
- Dimensions dictionary
- Global attributes dictionary
- Remove record numbers
- Update keyDict
- Units, Fill values, other metadata
- Change order of when writer is called
- Remove ExtractoObsData if possible
- Update CMakeLists.txt with environment variable
- Run test to make sure it runs (but will fail because of unupdated test reference)
- Update test reference file
- run unit tests making use of the target converter and verify that they pass. Make sure you verify the results by examining the file using ```ncdump```, ```h5dump```, ```HDFview```, or a similar utility.

## ZenHub organization

Please make sure that you have properly set:
- Labels: OBS3
- Milestone: OBS - October 2021
- Estimate: An estimate of how long this issue will take to resolve. 1 = half a day. 2 = full day.
- Epics: Modify converters to IODAv2 format
- Assignees: Anyone **working** on this issue. Anyone who is not working directly on the issue, but
  who should receive notifications, should be listed below.
- Reviewers: 

## Notifications

Fill in a list of GitHub handles here. People here are not *assigned*, but instead receive notifications
about this issue's progress.

Notifications should always go to: @rhoneyager.

Other people:

### PR DESCRIPTION
**Pull request by HamidehGMAO**
## Description

Upgrade Python converter for argoClim2ioda to IODA-v2

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #705

### CODE REVIEW

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
@HamidehGMAO , the dates in the output file are wrong. Also, not your doing but the "helper" at line 209 is wrong also .

**Comment by HamidehGMAO** [comment]
browsing on internet for argo climatology data, it seems the data is coming from "2004-2018 RG Argo Temperature Climatology. " file in https://sio-argo.ucsd.edu/RG_Climatology.html. Is that right @aerorahul ? it seems we have monthly average data from 2004 to 2018

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -388,12 +388,13 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_hfradar
...
 ecbuild_add_test( TARGET  test_${PROJECT_NAME}_argoclim
                   TYPE    SCRIPT
+                  ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
                   COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
                   ARGS    netcdf
                           "${Python3_EXECUTABLE} ${CMAKE_BINARY_DIR}/bin/argoClim2ioda.py
                           -i testinput/argoclim_test.nc
                           -o testrun/argoclim.nc
-                          -d 2019101600"
+                          -d 2004010200"
```
Comment (lliu1220) [inline test/CMakeLists.txt:397]: looks like the actual output time was marked as :
"2004-02-16T00:00:00Z"
"2004-03-16T00:00:00Z"
which is different from the -d 2004010200

Comment (HamidehGMAO) [reply test/CMakeLists.txt:397]: yeah, I know. I'm not sure what we should do for timing for this data. it's different from other type of data we have. the complete data file (you can find it here https://sio-argo.ucsd.edu/RG_Climatology.html) contains the average monthly data from 20040101 to end of 2018. Here just a couple of points of data has been saved as a sample test file. I'm not sure how the time is going to be used for these kind of data.

Comment (lliu1220) [reply test/CMakeLists.txt:397]: looks like so. Thank you for letting me know.

**Review by lliu1220 — APPROVED** [review]

File: test/CMakeLists.txt (hunk 1 above)
Comment (lliu1220) [reply test/CMakeLists.txt:397]: sorry, one last doubt- should ObsError all be zero?

Comment (HamidehGMAO) [reply test/CMakeLists.txt:397]: Well, The obs error is set to 0 and qc to 1 by @aerorahul  2 years ago, and as I know this converter has not been used so far. Maybe he can answer better. It's not very clear to me how and when we are going to use this data. Probably we need to revisit this converter later whn we are going to actually use it.

Comment (aerorahul) [reply test/CMakeLists.txt:397]: This is a converter of the Argo climatology.  It is not your typical observation dataset.  The intended purpose of this dataset was not for assimilation, but rather for providing a reference for comparison of the model and analysis departures from climatology.  

That being said, I am not sure what value would be appropriate for observation error, if this dataset was considered for assimilation.

Comment (lliu1220) [reply test/CMakeLists.txt:397]: ARGO_TEMPERATURE_ANOMALY
seems to be the obs error,
if using ARGO_TEMPERATURE_MEAN + ARGO_TEMPERATURE_ANOMALY as ObsValue

**Comment by srherbener** [comment]
@guillaumevernieres have your comments been addressed? Thanks

File: test/CMakeLists.txt (hunk 1 above)
Comment (ShastriPaturi) [reply test/CMakeLists.txt:434]: Agree with @lliu1220. Good point.
ARGO has a reported salinity drift in 2016 which was supposedly corrected in the climatology.

**Comment by guillaumevernieres** [comment]
> @guillaumevernieres have your comments been addressed? Thanks

@srherbener , yes, all good.

**Comment by srherbener** [comment]
> > @guillaumevernieres have your comments been addressed? Thanks
> 
> @srherbener , yes, all good.

Thanks @guillaumevernieres. Please approve (to get rid of the "changes requested" block) Thanks!

**Review by guillaumevernieres — APPROVED** [review]

**Pull request merged**